src/aiperf/dataset/composer/custom.py

```
@@ -25,14 +32,118 @@ def create_dataset(self) -> list[Conversation]:
-        utils.check_file_exists(self.config.input.file)
+        check_file_exists(self.config.input.file)
+
+        # Auto-infer dataset type if not provided
+        dataset_type = self.config.input.custom_dataset_type
+        if dataset_type is None:
+            dataset_type = self._infer_dataset_type(self.config.input.file)
+            if dataset_type is None:
+                raise ValueError(
+                    f"Could not infer dataset type from file: {self.config.input.file}. "
+                    "Please specify --custom-dataset-type explicitly."
+                )
```

Ruff (0.14.1)

Avoid specifying long messages outside the exception class

(TRY003)